### DISCUSSION TITLE
[v3] Laravel Echo does not work with v3

### DISCUSSION THREAD

**Discussion by yoeriboven** [livewire/livewire#5880] [7. Bugs (archived)] [answered]
Tried to upgrade my app to v3 but Livewire ignores Laravel Echo events. It works fine on v2.

Tried to do it again on a fresh install. Same issue.

Created a repo with instructions in which you can see it doesn't work.

[https://github.com/yoeriboven/livewire-echo](https://github.com/yoeriboven/livewire-echo)

**Comment by edalzell** [comment]
@yoeriboven I can't see that repo, is it private?

**Comment by joshhanley** [comment]
@yoeriboven yeah @edalzell is correct, repo is private. Can you make it public so we can have a look? Thanks!

**Comment by yoeriboven** [comment]
@edalzell @joshhanley Sorry about that. Must have happened automatically. It's public now!

**Comment by adesugbaa** [comment]
I have the same issue - my app went silent unable to join presence channels, and any other events/channels.

I noticed that the app is no longer calling the /broadcast/auth route after login

**Comment by yoeriboven** [comment]
@adesugbaa Interesting. I changed to public channels and it still doesn't work though.

**Comment by adesugbaa** [comment]
@yoeriboven correct. My application uses presence, private and public channels. After the upgrade, there is no event activity - the application opens a websocket and is completely radio silent afterwards. It is neither listening nor dispatching events.

**Comment by edalzell** [comment]
Ya I don't think the echo code is in there at all, a search for `echo:` returns nothing in v3, whereas it was there in v2.

This is something that still needs to be added, I guess

**Comment by joshhanley** [comment]
Yeah just to confirm @edalzell's findings, there is no implementation for V3 as it was overlooked but we will work on getting that sorted out 🙂

**Comment by adesugbaa** [comment]
Will there be an issue that tracks this missing feature?

**Comment by edalzell** [comment]
Added here: https://github.com/livewire/livewire/pull/6092

**Comment by calebporzio** [comment, marked as answer]
Just added it back (oops haha that's why we're in beta) - will be available in the next release #6092

**Reply by wizzymore** [reply]
Testing the change using the main branch it seems that now it listens for the echo events but how do you get the event data in v3?

Unable to resolve dependency [Parameter #0 [ <required> $event ]] in class App\Livewire\Content when i try the method that works in v2
```public function appendContent($event): void```

**Reply by wizzymore** [reply]
<strike>
@calebporzio after a bit of investigation seems like the bug is caused by dispatching objects.
My event is the following structure:
```
{
  content: string;
  rawContent: string;
}
```

Whenever I send an object from javascript to livewire that can't be converted to a php array ( using numbers for keys ) i don't receive any data in the function. So i think it's something wrong with the serialization.
</strike>

I just realised that i need to add all the params like this:
```public function appendContent(string $content, string $rawContent):```
And now i have access to both of them
